Some GUI hacks for better UX

emagra opened the pull request:
Control buttons (Random...) position fixed on scroll. Always visible while scrolling long playlist
Hide Love and external player, show volume slider (I think more useful)
Didn't find a better way to avoid moving time in the list on mouse over to display the bin. Just add spaces in the column.

emagra committed:
Control buttons (Random...) position fixed on scroll
Hide Love and external player, show volume slider

SuperBFG7 commented:
Hi Emanuele

Thanks for your pull request.
I like fixing the control buttons, I would definitely merge this.

About the love button/external player vs. volume slider:
- Does the volume slider work? I was under the impression it didn't actually work, but as I never used it I can't tell (my DAC doesn't support volume control)
- I am using the love button, so I would like to keep it. One option would be to make is only visible, when the channel to communicate with mpdas is actually available. Never got around to implement this tough.
- The external player I don't really use anymore, for me it would be fine to remove it completely. Unless other people are still using it (anyone?).

Finally about avoiding the table width to change when the bin is displayed, I suggest inserting a div, e.g.:
`<div style="width: 14px;"></div>`

I don't like adding non-breaking spaces and you also added an additional header column that should not be there.

emagra commented:
Hello there,

> Does the volume slider work? I was under the impression it didn't actually work, but as I never used it I can't tell (my DAC doesn't support volume control)

for me the volume slider works like a charm (my setup is ubuntu server with mpd no much change on default settings), and I'd like to have also a number to display the volume within the slider.

> I am using the love button, so I would like to keep it. One option would be to make is only visible, when the channel to communicate with mpdas is actually available. Never got around to implement this tough.
The external player I don't really use anymore, for me it would be fine to remove it completely. Unless other people are still using it (anyone?).

Love button and external player doesn't work for me. Didn't investigate much but I think something about user permission. Maybe the love button can be placed in the song info box, because is more related with the song then the control.

Something like this:
![immagine](https://user-images.githubusercontent.com/149369/121223912-4f829f00-c888-11eb-9a6c-01e5dfa1e73e.png)


> 
> Finally about avoiding the table width to change when the bin is displayed, I suggest inserting a div, e.g.:
> <div style="width: 14px;"></div>
Yeah definitely better. I will set to 15px cause on chrome I see a little bit of movement.

emagra committed:
Better UX
1. hide external player
2. move love in current track info and make it red 'cause love

emagra committed:
move the love button to the right (forgot in the prev commit)

emagra committed:
Volume slider with the number on right

emagra commented:
Hello again,
make 2 commit (1 'cause I forget the float right) 

1. adapt the code to your comment
2. add the number in volume slider. Don't like the result very much but is what i'd like

The love button for me doesn't work cause I use  [mpdscribble](https://github.com/MusicPlayerDaemon/mpdscribble), and from the docs to send some love the command is `sendmessage mpdscribble love`. Don't know how much different from mpdas

emagra committed:
remove "root" in path is ugly

emagra committed:
need something so replace root with Library

emagra committed:
Separate the song info box from playlist/library list and position fixed for song info box

SuperBFG7 commented:
Hi Emanuele
while merging your changes, I added some additional changes and ended up with a couple of individual commits instead of a simple merge. I hope this is ok and will close the pull request now.

- Removed the standalone player (I don't consider this usefull anymore): d5294675c4dccfc36fc2d965c44b9186f1308aa5
- As suggested I separated the song info box and the queue (886e390d0995225ccd786a85df37cddb62ccdfc4). The styles I moved from the HTML to the CSS file.
- To keep the UI useable also on small screens (mobile phone), I limited the stickiness of the song info box to big screens only: 86586fbcee0944e77fa32e17c5d0155e37807df3
- Enabled the volume slider again and added the volume number (28844275e622fd2be5c590e3df4ea1128c6965b7) but display it only when volume control is supported by the output (4d30fff4bc598c68848eead86d5f90324039b2a7)
- Renamed root to Library: 8669750a4befcde769e9cb2a5f8c3ce4920f9c18
- Moved love button to the song info box and color it red (I love this change, much better than in the controls): 28844275e622fd2be5c590e3df4ea1128c6965b7
- Added mpdscribble support and only display the love button when either mpdas or mpdscribble are active: 2b461aafa2d64488f862fb9e422b416320c952b4

Let me know if the mpdscribble support works (couldn't test this).

cheers!
Daniel

emagra commented:
Yeah. Compile the master branch and I really like it. Better with the commits, I messed up a little bit with the code and in the pull request. 
The love button doesn't seems to work with mpdscribble and lastfm (should set love song on the site right?) anyway no error message no info. When I have a little bit of time I'll try to see.

Cheers!

SuperBFG7 commented:
Yes, it should let Last.fm know that you love this song. Strange it doesn’t work. Because if you see the love button, the mpdscribble channel is present and all ympd does is send a message (love) on the channel. Same as doing it via mpc. Does it work with mpc?
Am 21. Juni 2021, 22:38 +0200 schrieb Emanuele Grande ***@***.***>:
> Yeah. Compile the master branch and I really like it. Better with the commits, I messed up a little bit with the code and in the pull request.
> The love button doesn't seems to work with mpdscribble and lastfm (should set love song on the site right?) anyway no error message no info. When I have a little bit of time I'll try to see.
> Cheers!
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub, or unsubscribe.

emagra commented:
Yeah the problem seems to be mpdscribble. 

`$: mpc channels `
`mpdscribble`
`$: mpc sendmessage mpdscribble love`
`: `

channel is there, sending message from terminal give no output and doesn'twork. Using `mpc waitmessage mpdscribble` and clicking the love button the message is shown. Trying to see in mpdscribble github repo.

Thanks and cheers!!

> Yes, it should let Last.fm know that you love this song. Strange it doesn’t work. Because if you see the love button, the mpdscribble channel is present and all ympd does is send a message (love) on the channel. Same as doing it via mpc. Does it work with mpc? Am 21. Juni 2021, 22:38 +0200 schrieb Emanuele Grande ***@***.***>:
> […](#)
> Yeah. Compile the master branch and I really like it. Better with the commits, I messed up a little bit with the code and in the pull request. The love button doesn't seems to work with mpdscribble and lastfm (should set love song on the site right?) anyway no error message no info. When I have a little bit of time I'll try to see. Cheers! — You are receiving this because you modified the open/close state. Reply to this email directly, view it on GitHub, or unsubscribe.
